File: src/main/java/fi/quanfoxes/parser/Constructor.java
package fi.quanfoxes.parser;

import fi.quanfoxes.AccessModifier;

public class Constructor extends Function {
    private static final String IDENTIFIER = "constructor";
    private static final String INDEXED_IDENTIFIER = "constructor_%d";

    private boolean automatic;

    public static Constructor empty(Context context) {
        Constructor constructor = new Constructor(context, AccessModifier.PUBLIC, true);
        constructor.setParameters(new Node());

        return constructor;
    }

    public Constructor(Context context, int modifiers) {
        this(context, modifiers, false);
    }

    public Constructor(Context context, int modifiers, boolean automatic) {
        super(context, modifiers);
        this.automatic = automatic;
    }

    public boolean isDefault() {
        return automatic;
    }

    @Override
    public String getIdentifier() {
        int index = getIndex();

        if (index != -1) {
            return String.format(INDEXED_IDENTIFIER, index);
        }
        else {
            return IDENTIFIER;
        }
    }
}

File: Zigzag/Parser/Constructor.cs
using System.Collections.Generic;

public class Constructor : Function
{
	public bool IsDefault { get; private set; }
	public bool IsEmpty => IsDefault && (GetTypeParent()!.Initialization?.IsEmpty ?? true);

	public static Constructor Empty(Context context)
	{
		return new Constructor(context, AccessModifier.PUBLIC, new List<Token>(), true);
	}

	public Constructor(Context context, int modifiers, List<Token> blueprint, bool is_default = false) : base(context, modifiers, "init", blueprint)
	{
		IsDefault = is_default;
	}
}